### PR TITLE
CEO#4.9 - Game.ts

### PR DESCRIPTION
**Pull request by d-little**
### Game.ts

No drafting, that can come later. Just the deck + hand dealing so we can get testing first.

### CODE REVIEW

File: src/server/SerializedGame.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ export type SerializedGame = {
     aresData?: AresData;
     awards: Array<string>;
     board: SerializedBoard;
+    ceoDeck: SerializedDeck;
```
Comment (kberg) [inline src/server/SerializedGame.ts:23]: For many games, this will be undefined to start with. Without care, this kind of thing makes it possible for the server to throw errors while loadingr legacy games.

The standard way we've dealt with that is by changing this to `ceoDeck?: SerializedDeck;`. And then in `deserialize` or here, add a comment "Remove the conditional by..." and pick a date at least 2 weeks from now. Maybe 4.

File: tests/cards/ceo/Bjorn.spec.ts [hunk 2]
```diff
@@ -17,7 +17,9 @@ describe('Bjorn', function() {
 
   beforeEach(() => {
     card = new Bjorn();
-    game = newTestGame(4, {ceoExtension: true});
+    // Todo: Prelude extension is only activated here as we didnt have enough CEOs in early testing
```
Comment (kberg) [inline tests/cards/ceo/Bjorn.spec.ts:20]: nit: capitalize TODO. Especially since you're going to change code anyway.

File: tests/cards/ceo/Will.spec.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@ describe('Will', function() {
 
   beforeEach(() => {
     card = new Will();
-    game = newTestGame(4, {ceoExtension: true});
+    game = newTestGame(1, {ceoExtension: true});
```
Comment (kberg) [inline tests/cards/ceo/Will.spec.ts:24]: Why the change from 4 to 1?

**Review by kberg — COMMENTED** [review]
Much easier to review without the draft. Much easier.

File: tests/cards/ceo/Will.spec.ts (hunk 3 above)
Comment (d-little) [reply tests/cards/ceo/Will.spec.ts:24]: We only have one test player here.

CEOs cannot currently support 4 players without expansions, we dont have enough in the base game.  Cleaner to switch to one player.

Now that I think about it, does this matter for Solo vs. Multiplayer game testing?  For this example I dont think it should, but some CEOs might care if there are no opponents

File: src/server/SerializedGame.ts (hunk 1 above)
Comment (d-little) [reply src/server/SerializedGame.ts:23]: Remove Conditional By?  Or After date x?

Comment (kberg) [reply src/server/SerializedGame.ts:23]: Doesn't matter. We'll know.

File: tests/cards/ceo/Will.spec.ts (hunk 3 above)
Comment (kberg) [reply tests/cards/ceo/Will.spec.ts:24]: /shrug you tell me!

File: src/server/SerializedGame.ts (hunk 1 above)
Comment (d-little) [reply src/server/SerializedGame.ts:23]: Ah, making it optional freaks out CeoDeck.deserialize() because it could come through as undefined.

I _think_ it should be handled with some temporary code that looks like this?  Maybe?  I'm just not sure what to put in the `{}`, it needs to be type SerializedDeck

```ts
    // TODO(dl): remove ?? {} by 2023/03/20
    const ceoDeck = CeoDeck.deserialize(d.ceoDeck ?? {}, rng);

```

Comment (kberg) [reply src/server/SerializedGame.ts:23]: SerializedDeck is of this type:

```
export type SerializedDeck = {
  drawPile: Array<CardName>;
  discardPile: Array<CardName>;
}
```

So the empty deck is `{drawPile: [], discardPile: []}`

**Review by kberg — APPROVED** [review]
